model.md

```diff
@@ -0,0 +1,86 @@
+|        |                |                          |                    | studentId not found                               | throw error ('student not found')                                  |
+|        |                |                          |                    | cohortId not found                                | throw error ('cohort not found')                                   |
+|        |                |                          |                    | cohortStudentSize >= cohort.capacity              | throw error ('cohort capacity exceeded')                           |
+|        |                |                          |                    |                                                   |                                                                    |
+|        |                | removeStudentFromCohort  | studentId:number,  | inputIds match existing Ids                       | foundStudent.cohortId = null, return foundStudent                  |
+|        |                |                          | cohortId:number    |                                                   |                                                                    |
+|        |                |                          |                    | studentId not found                               | throw error ('studentId not found')                                |
+|        |                |                          |                    | cohortId not found                                | throw error ('cohortId not found')                                 |
+|        |                |                          |                    |                                                   |                                                                    |
+|        |                |                          |                    | foundStudent.cohortId !== inputCohortId           | throw error ('student not in passed cohort')                       |
+|        |                |                          |                    |                                                   |                                                                    |
+|        |                |                          |                    |                                                   |                                                                    |
+|        | Student        |                          |                    |                                                   |                                                                    |
+|        |                | details                  |                    |                                                   | return { ... student properties }                                  |
+|        |                |                          |                    |                                                   |                                                                    |
+
+## CohortManager
```

Comment:
Very smart to just have the properties separate, really made a mess for myself by putting it into one table

Reply:
Thanks, yeah the one table was going to get very cramped if I had even more columns in it.